### PR TITLE
feat(export): add support for usage analytics export

### PR DESCRIPTION
**Pull request by subratsahilgupta**
## Summary by CodeRabbit

* **New Features**
  * Export usage analytics to CSV (one row per analytics item) with optional customer metadata columns.
  * New scheduled export type: usage_analytics.
* **Validation / Bug Fixes**
  * Metadata fields are validated against the chosen export type; unsupported types rejected and missing column names defaulted.
  * Metadata validation applied during export/job creation.
* **Tests**
  * Added tests for usage analytics export, metadata-driven columns, and edge cases (headers-only, missing metadata).

### CODE REVIEW

File: internal/service/sync/export/usage_analytics_export.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		"env_id", request.EnvID,
+		"start_time", request.StartTime,
+		"end_time", request.EndTime)
+
+	ctx = types.SetTenantID(ctx, request.TenantID)
+	ctx = types.SetEnvironmentID(ctx, request.EnvID)
+
+	customers, err := e.customerRepo.ListAll(ctx, &types.CustomerFilter{
+		QueryFilter: types.NewNoLimitQueryFilter(),
+	})
```
Comment (nkmishra1997) [inline internal/service/sync/export/usage_analytics_export.go]: Rather than this - we need to prepare only for external_Customer_ids for which we got > 0 event in the said start end time range

File: internal/service/sync/export/usage_analytics_export.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+			e.logger.Warnw("failed to fetch usage analytics for customer, skipping",
+				"customer_id", c.ID,
+				"external_id", c.ExternalID,
+				"error", err)
+			continue
+		}
+
+		for _, item := range response.Items {
+			record := &usageAnalyticsRecord{
+				CustomerID:         c.ID,
```
Comment (nkmishra1997) [inline internal/service/sync/export/usage_analytics_export.go:177]: remove

File: internal/service/sync/export/usage_analytics_export.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+			StartTime:          request.StartTime,
+			EndTime:            request.EndTime,
+		})
+		if err != nil {
+			e.logger.Warnw("failed to fetch usage analytics for customer, skipping",
+				"customer_id", c.ID,
+				"external_id", c.ExternalID,
+				"error", err)
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (subratsahilgupta) [reply internal/service/sync/export/usage_analytics_export.go:173]: we should not stop export if one fetch on customer fails

File: internal/service/sync/export/usage_analytics_export.go (hunk 2 above)
Comment (subratsahilgupta) [reply internal/service/sync/export/usage_analytics_export.go:177]: no need to

File: internal/repository/clickhouse/event.go [hunk 4]
```diff
@@ -1276,3 +1276,61 @@ func (r *EventRepository) GetEventByID(ctx context.Context, eventID string) (*ev
 	SetSpanSuccess(span)
 	return &event, nil
 }
+
+// GetDistinctEventNames retrieves distinct event names for the given external customer IDs
+func (r *EventRepository) GetDistinctExternalCustomerIDs(ctx context.Context, startTime, endTime time.Time) ([]string, error) {
```
Comment (omkar273) [inline internal/repository/clickhouse/event.go:1281]: can add req `dto` instead of positional args since going forward it will make adding new params and their validation difficult, requiring to update all caller sites and testing it again

Comment (subratsahilgupta) [reply internal/repository/clickhouse/event.go:1281]: NIT

File: internal/service/sync/export/usage_analytics_export.go [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+	csvBytes := buf.Bytes()
+
+	if recordCount == 0 {
+		e.logger.Infow("no usage analytics data found for export - uploading empty CSV with headers only",
+			"tenant_id", request.TenantID,
+			"env_id", request.EnvID,
+			"csv_size_bytes", len(csvBytes))
+	} else if failedCount > 0 {
+		e.logger.Warnw("usage analytics export completed with partial data",
```
Comment (nkmishra1997) [inline internal/service/sync/export/usage_analytics_export.go:215]: in follow up : change all warn logs in the codebase to error logs

**Review by nkmishra1997 — APPROVED** [review]
